### PR TITLE
fix: unify error serialization with config and other values. Fix err.onFai…

### PR DESCRIPTION
**Pull request by AtofStryker**
…l issues

### User facing changelog

n/a

### Additional details
The goal of this PR is to fix the `err.onFail` is not a function bug we are seeing while writing multi-domain unit tests, as well as unify the error serialization across domains. 

`err.onFail` was failing due to non serializable properties being stringified back to the primary domain, causing the error to fail. Instead of trying to send non serializable values as a serialized string, we now just omit these properties before sending the error (similar to env and config). Now that we process errors in the `ran:domain:fn` context, we need to also account for scenarios when `throw 'oops'` or something is thrown in the secondary that may/may not be an error object. This PR should cover this at a basic level.

I would have liked to use the `structuredClone` algorithm to clone Errors where applicable, but interestingly all properties declared on the error are removed when trying to clone, either with native Chrome implementations or the ponyfill. Once the Error is sent through `postMessage`, it loses it's prototype hierarchy and ultimately just becomes an object. Therefor, I figured the method we had to pre/post process errors how we do it now is likely preferable. 

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/driver/src/multi-domain/communicator.ts [hunk 1]
```diff
@@ -132,7 +103,7 @@ export class SpecBridgeDomainCommunicator extends EventEmitter {
 
     try {
       // We always want to make sure errors are posted, so clean it up to send.
-      const preProcessedError = preprocessErrorForPostMessage(err)
+      const preProcessedError = _.isError(err) ? preprocessErrorForPostMessage(err) : err
```
Comment (AtofStryker) [inline packages/driver/src/multi-domain/communicator.ts:106]: This becomes a problem if someone tries to throw something that is unserializable. Do we think this is a valid concern?

**Comment by mjhenkes** [comment]
It might be good to hold off on this PR until after #20247 lands.

File: packages/driver/src/multi-domain/communicator.ts (hunk 1 above)
Comment (mjhenkes) [reply packages/driver/src/multi-domain/communicator.ts:106]: I think it is a valid concern. In general i think we should add a basic try-catch around postMessage for any unserializable data, so we can at least send a message with an error.

**Comment by AtofStryker** [comment]
going to move this to draft until #20247 is merged in

**Comment by AtofStryker** [comment]
closing in favor of #20520

**Pull request closed without merging**